File: SimpleApp/MainActivity.cs
namespace SimpleApp;

[Activity(Label = "@string/app_name", MainLauncher = true)]
public class MainActivity : Activity
{
    protected override void OnCreate(Bundle? savedInstanceState)
    {
        base.OnCreate(savedInstanceState);

        // Set our view from the "main" layout resource
        SetContentView(Resource.Layout.activity_main);

        // Get references to the UI elements
        var textView = FindViewById<TextView>(Resource.Id.textView);
        var button = FindViewById<Button>(Resource.Id.button);

        // Set up the button click handler
        if (button != null && textView != null)
        {
            button.Click += (sender, e) =>
            {
                textView.Text = "Hello from Xamarin!";
            };
        }
    }
}

File: EspressoTests/src/main/java/com/companyname/simpleapp/MainActivity.java
package com.companyname.simpleapp;

import android.app.Activity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends Activity {
    
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        
        // Get references to the UI elements
        final TextView textView = findViewById(R.id.textView);
        Button button = findViewById(R.id.button);
        
        // Set up the button click handler
        if (button != null && textView != null) {
            button.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    textView.setText("Hello from Xamarin!");
                }
            });
        }
    }
}
